Makefile

```diff
@@ -83,26 +83,22 @@ bin/Test$(CONFIGURATION)/$(NATIVE_TIMING_LIB): tests/NativeTiming/timing.c $(wil
 	mkdir -p `dirname "$@"`
 	gcc -g -shared -o $@ $< -m32 $(JI_JDK_INCLUDE_PATHS:%=-I%)
 
-bin/Test$(CONFIGURATION)/Java.Interop-Tests.dll: $(wildcard src/Java.Interop/*/*.cs src/Java.Interop/Tests/*/*.cs)
+# Usage: $(call TestAssembly,assembly-basename)
+define TestAssembly
+bin/Test$(CONFIGURATION)/$(1)-Tests.dll: $(wildcard src/$(1)/*/*.cs src/$(1)/Test*/*/*.cs)
 	$(XBUILD)
 	touch $@
+endef
 
-bin/Test$(CONFIGURATION)/Java.Interop.Dynamic-Tests.dll: $(wildcard src/Java.Interop.Dynamic/*/*.cs src/Java.Interop.Dynamic/Tests/*/*.cs)
-	$(XBUILD)
-	touch $@
-
-bin/Test$(CONFIGURATION)/Java.Interop.Export-Tests.dll: $(wildcard src/Java.Interop.Export/*/*.cs src/Java.Interop.Export/Tests/*/*.cs)
-	$(XBUILD)
-	touch $@
+$(eval $(call TestAssembly,Java.Interop))
+$(eval $(call TestAssembly,Java.Interop.Dynamic))
+$(eval $(call TestAssembly,Java.Interop.Export))
+$(eval $(call TestAssembly,Java.Interop.Tools.JavaCallableWrappers))
 
 bin/Test$(CONFIGURATION)/Java.Interop-PerformanceTests.dll: $(wildcard tests/Java.Interop-PerformanceTests/*.cs) bin/Test$(CONFIGURATION)/$(NATIVE_TIMING_LIB)
 	$(XBUILD)
 	touch $@
 
-bin/Test$(CONFIGURATION)/Android.Interop-Tests.dll: $(wildcard src/Android.Interop/*/*.cs src/Android.Interop/Tests/*/*.cs)
```

Comment:
Is this related? Should this be added to the list of assemblies above? Else maybe it should be moved to a separate commit?